lib/RestructureCFG/GenericRegionPass.cpp

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include "revng-c/RestructureCFG/GenericRegionInfo.h"
+#include "revng-c/RestructureCFG/GenericRegionPass.h"
+
+char GenericRegionPass::ID = 0;
+
+static constexpr const char *Flag = "generic-region-info";
+using Reg = llvm::RegisterPass<GenericRegionPass>;
+static Reg X(Flag, "Perform the generic region identification analysis");
+
+bool GenericRegionPass::runOnFunction(llvm::Function &F) {
+
+  // Run the `GenericRegionInfo`
+  GenericRegionInfo<llvm::Function *> RIA;
+  RIA.compute(&F);
+
+  // The goal of `RegionIdentificationPass` is to just perform an analysis
+  // which does not perform changes to the IR
+  return false;
+}
+
+void GenericRegionPass::getAnalysisUsage(llvm::AnalysisUsage &AU) const {
+
+  // This is a read only analysis, that does not touch the IR
+  AU.setPreservesAll();
+}
+
+void GenericRegionPass::getResult() {
+  return;
+}
```

Comment:
Fixed here 

https://github.com/revng/revng-c/blob/2874031652f8d9c4240ed2796912c23c75054aca/lib/RestructureCFG/GenericRegionPass.cpp#L32

Reply:
I'd like it better to return a `const &`, to prevent passes depending on this to mutate it.

All the rest is perfect.

I'll close the other related comments, and leave just this one open, until this is fixed.

Reply:
Right, fixed this here

https://github.com/revng/revng-c/blob/e211fe822643ea506c2fccf69b773fb21cf9065c/include/revng-c/RestructureCFG/GenericRegionPass.h#L26

Reply:
:+1: